alsrb0504/20230418/욕심쟁이 판다/욕심쟁이 판다.js

```diff
@@ -0,0 +1,63 @@
+// const readFileSyncAddress = "/dev/stdin";
+const readFileSyncAddress = "text.txt";
+
+const input = require("fs")
+  .readFileSync(readFileSyncAddress)
+  .toString()
+  .trimEnd()
+  .split("\n");
+
+const dy = [0, 0, -1, 1];
+const dx = [-1, 1, 0, 0];
+
+const N = Number(input[0]);
+const map = input.slice(1, 1 + N).map((el) => el.split(" ").map(Number));
+// DP 맵을 0으로 초기화
+const dp = Array.from({ length: N }, () => new Array(N).fill(0));
+let answer = 0;
+
+// DFS를 하면서 DP 배열의 값을 채움.
+// 만약 DP 배열의 값이 0이면 DFS 진행
+for (let i = 0; i < N; i++) {
+  for (let j = 0; j < N; j++) {
+    if (dp[i][j] === 0) dfs(i, j, 1);
+  }
+}
+
+// 결과값이 저장된 DP 배열 중 가장 큰 값을 찾음
+for (let i = 0; i < N; i++) {
+  for (let j = 0; j < N; j++) {
+    answer = Math.max(answer, dp[i][j]);
+  }
+}
```

Comment:
```suggestion
for (let i = 0; i < N; i++) {
  for (let j = 0; j < N; j++) {
    if (dp[i][j] === 0) answer = Math.max(answer, dfs(i, j, 1));
  }
}
```

dfs 함수를 보면 1이나 max 값이 반환되는 것 같은데 그렇다면!!
이중 for문 하나로 그때그때 answer을 재설정해줄 수도 있을 것 같아요

Reply:
오.. 맞네요 확실히 2번 반복하지 않고도 구할 수 있었는데.. 좋은 지적 감사합니다!